controllers/helmrepository_controller_oci.go

```diff
@@ -366,3 +386,42 @@ func (r *HelmRepositoryOCIReconciler) eventLogf(ctx context.Context, obj runtime
 	}
 	r.Eventf(obj, eventType, reason, msg)
 }
+
+// oidcAuth generates the OIDC credential authenticator based on the specified cloud provider.
+func oidcAuth(ctx context.Context, obj *sourcev1.HelmRepository) (helmreg.LoginOption, error) {
+	url := strings.TrimPrefix(obj.Spec.URL, sourcev1.OCIRepositoryPrefix)
+	ref, err := name.ParseReference(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse URL '%s': %w", obj.Spec.URL, err)
+	}
+
+	loginOpt, err := loginWithManager(ctx, obj.Spec.Provider, url, ref)
+	if err != nil {
+		return nil, fmt.Errorf("failed to login to registry '%s': %w", obj.Spec.URL, err)
+	}
+
+	return loginOpt, nil
+}
```

Comment:
The code in this function combined with `loginWithManager()` below, it seems like `registry.OIDCAdaptHelper()` can just take the result of `OCIRepositoryReconciler.oidcAuth()`. The arguments of `OCIRepositoryReconciler.oidcAuth()` can just be the URL and the provider name to not make it source type specific. It should be okay to convert `OCIRepositoryReconciler.oidcAuth()` into an independent function `oidcAuth()`.

In the future, I think we can make a private package with a lot of the OCI related code here and in OCIRepo reconciler.

Reply:
true, it would be better. But `OCIRepositoryReconciler.oidcAuth()` is a receiver function, so I thought that this would need a specific PR to refactor it.

Reply:
okay to change it separately.